src/amp-story-player/amp-story-player-impl.js

```diff
@@ -309,9 +313,25 @@ export class AmpStoryPlayer {
     this.initializeIframes_();
     this.initializeButton_();
     this.signalReady_();
+    if (this.isAutoplaying_) {
+      this.setUpAutoplayClickShield_();
+    }
     this.isBuilt_ = true;
   }
 
+  /**
+   * Sets up a listener disable autoplay mode on
+   * the first user interaction.
+   * @private
+   */
+  setUpAutoplayClickShield_() {
+    this.element_.addEventListener(
+      'click',
+      () => this.toggleAutoplayMode_(false),
+      {once: true}
```

Comment:
What would happen in the following case?

player in "preview" -> user clicks -> player exits out of "preview" -> user exits player

i think we would want the player to toggle back to "preview" mode, right? if so, then we would need this listener to persist, and not add/remove on every toggle.

Reply:
I see. The click shield already "shows" automatically when the player has the autoplay attribute, so making the listener persist would make it work the same every time.

The only issue is there is no way to re-set the autoplay mode (which I think should be up to the publisher)